pkg/virt-controller/watch/clone/clone.go

```diff
@@ -431,6 +498,62 @@ func (ctrl *VMCloneController) logAndRecord(vmClone *clonev1alpha1.VirtualMachin
 	log.Log.Object(vmClone).Infof(msg)
 }
 
+func (ctrl *VMCloneController) getTargetType(vmClone *clonev1alpha1.VirtualMachineClone) cloneTargetType {
+	if vmClone.Spec.Target != nil {
+		return cloneTargetType(vmClone.Spec.Target.Kind)
+	} else {
+		return defaultType
+	}
+}
+
+func (ctrl *VMCloneController) getSource(vmClone *clonev1alpha1.VirtualMachineClone, name, namespace, sourceKind string, store cache.Store) (interface{}, error) {
+	key := getKey(name, namespace)
+	obj, exists, err := store.GetByKey(key)
+	if err != nil {
+		return nil, fmt.Errorf("error getting %s %s in namespace %s from cache: %v", sourceKind, name, namespace, err)
+	}
+	if !exists {
+		err = ctrl.updateStatus(vmClone, syncInfoType{
+			isCloneFailing: true,
+			failEvent:      SourceDoesNotExist,
+			failReason:     fmt.Sprintf("%s %s does not exist in namespace %s", sourceKind, name, namespace),
+		})
+
+		if err != nil {
+			log.Log.Errorf("updating status when source %s does not exist failed: %v", sourceKind, err)
+		}
+
+		return nil, fmt.Errorf("%s %s in namespace %s does not exist", sourceKind, name, namespace)
+	}
+
+	return obj, nil
+}
+
+func (ctrl *VMCloneController) getVmFromSnapshot(snapshot *snapshotv1alpha1.VirtualMachineSnapshot) (*k6tv1.VirtualMachine, error) {
+	var contentName string
+	if snapshot.Status != nil && snapshot.Status.VirtualMachineSnapshotContentName != nil {
```

Comment:
There is a function for this: getVMSnapshotContentName make it public and use it instead. like with 
VmRestoreProgressing and VmSnapshotReady